ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp

```diff
@@ -223,16 +223,30 @@ Tensor ExecuteDiv::invoke(
         "Incorrect rounding mode (expected None, 'trunc', or 'floor')");
     output_tensor = output_tensor.value_or(ttnn::empty_like(input_a));
     auto arch = input_a.device()->arch();
-    if (arch == tt::ARCH::WORMHOLE_B0) {
+    if (arch != tt::ARCH::GRAYSKULL) {
         DataType input_dtype = input_a.get_dtype();
+
+        // No accurate_mode for FP32 div as inf/nan are handled at kernel level
+        if (input_dtype == DataType::FLOAT32 && input_b.get_dtype() == DataType::FLOAT32) {
+            Tensor result = ttnn::divide(queue_id, input_a, input_b, std::nullopt, output_mem_config, output_tensor);
+            if (round_mode == "trunc") {
+                result = ttnn::trunc(queue_id, result, output_mem_config, output_tensor);
+            } else if (round_mode == "floor") {
+                result = ttnn::floor(queue_id, result, output_mem_config, output_tensor);
+            }
+            return result;
+        }
+
         Tensor a = typecast(queue_id, input_a, DataType::FLOAT32);
         Tensor b = typecast(queue_id, input_b, DataType::FLOAT32);
-        Tensor result = ttnn::divide(queue_id, a, b);
+        // div without inf/nan handling
+        Tensor result =
+            ttnn::multiply(queue_id, a, ttnn::reciprocal(b), std::nullopt, output_mem_config, output_tensor);
 
```

Comment:
Do not forget to add the comments here about the side effect of how reciprocal is handling the zeros as you described that we are depending on.  Thanks for the fix here.